core-kotlin-modules/core-kotlin-7/src/main/kotlin/com/baeldung/optional/Optional.kt

```diff
@@ -0,0 +1,33 @@
+package com.baeldung.optional
+
+import java.util.Optional
+import kotlin.jvm.optionals.getOrDefault
+import kotlin.jvm.optionals.getOrElse
+import kotlin.jvm.optionals.getOrNull
+
+fun getOrNull() {
+    val optionalValue: Optional<String> = Optional.of("Hello, Baeldung!")
+    val value: String? = optionalValue.getOrNull()
+}
+
+fun getOrElse() {
+    val optionalValue: Optional<String> = Optional.empty()
+    val value: String = optionalValue.getOrElse { throw IllegalStateException("Value is missing") }
```

Comment:
This one we should wrap with the `assertThatThrows()` or equivalent when we convert to a unit test.